core/config/project_settings.cpp

```diff
@@ -923,6 +918,34 @@ Error ProjectSettings::_save_custom_bnd(const String &p_file) { // add other par
 Error ProjectSettings::save_custom(const String &p_path, const CustomMap &p_custom, const Vector<String> &p_custom_features, bool p_merge_with_current) {
 	ERR_FAIL_COND_V_MSG(p_path == "", ERR_INVALID_PARAMETER, "Project settings save path cannot be empty.");
 
+	PackedStringArray project_features = get("application/config/features");
+	// If there is no feature list currently present, force one to generate.
+	if (project_features.is_empty()) {
+		project_features = ProjectSettings::get_required_features();
+	}
+	// Check the rendering API.
+	const String rendering_api = get("rendering/quality/driver/driver_name");
+	if (rendering_api != "") {
+		// Add the rendering API as a project feature if it doesn't already exist.
+		if (!project_features.has(rendering_api)) {
+			project_features.append(rendering_api);
+		}
+	}
+	// Check for the existence of a csproj file.
+	if (FileAccess::exists(get_resource_path().plus_file(String(get("application/config/name")) + ".csproj"))) {
```

Comment:
That's clever, though might need double checking that it's the exact same code used to generate the `.csproj` (I can imagine there's a validation step as some characters allowed in the project name might not be allowed in file names).

I guess that also means you might not need to rely on `MODULE_MONO_ENABLED` anymore?

Reply:
(I see this is code just moved around, I missed it in the original PR. So this doesn't need to be changed in this hotfix PR.)

Reply:
`MODULE_MONO_ENABLED` is used to check if the current build of Godot supports C#, but this code is checking if a project has a C# project file (I do need to double check that it's the same logic).

Reply:
It wasn't the same logic. I updated the PR to have a copy of the logic from the Mono module. We might want to make this a method on ProjectSettings and use it in Mono too. I guess it's kind of worth doing in a separate PR but since it's urgent I think that I've talked myself into doing it in this PR...